### PR TITLE
Change for cipheriv

### PR DESCRIPTION
**Pull request by mhamann**
Replacement of #316

### CODE REVIEW

File: lib/nconf/stores/file.js [hunk 1]
```diff
@@ -185,7 +179,16 @@ File.prototype.stringify = function (format) {
...
   if (this.secure) {
-    data = this.keys.encrypt(data);
+    var self = this;
+    data = Object.keys(data).reduce(function (acc, key) {
+      var value = format.stringify(data[key]);
+      var iv = crypto.randomBytes(16);
+      var cipher = crypto.createCipheriv(self.secure.alg, self.secure.secret, iv);
+      var ciphertext = cipher.update(value, 'utf8', 'hex');
+      ciphertext += cipher.final('hex');
+      acc[key] = { alg: self.secure.alg, value: ciphertext, iv: iv.toString('hex') };
```
Comment (bnoordhuis) [inline lib/nconf/stores/file.js]: You should also store the authentication tag when `alg` is a CCM, GCM or OCB cipher (and store the full 128 bits, not one of the shorter variants.)

File: lib/nconf/stores/file.js [hunk 2]
```diff
@@ -199,11 +202,31 @@ File.prototype.stringify = function (format) {
...
   var parsed = this.format.parse(contents);
 
-  if (!this.secure) {
-    return parsed;
+  if (this.secure) {
+    var self = this;
+    var outdated = false;
+    parsed = Object.keys(parsed).reduce(function (acc, key) {
+      var value = parsed[key];
+      var decipher = crypto.createDecipher(value.alg, self.secure.secret);
```
Comment (bnoordhuis) [inline lib/nconf/stores/file.js]: You're calling `crypto.createDecipher()` unconditionally here, then conditionally do the work again when an IV is detected. That means you're doing twice the work.

Worse, it'll emit warnings with Node.js versions where `crypto.createDecipher()` is deprecated (v11.x and up.)

**Comment by mhamann** [comment]
@bnoordhuis mind submitting a PR with some fixups?

**Comment by bnoordhuis** [comment]
No thanks. My last PR was open for over a year, I reworked it at your request and then months later you close it in favor of your own changes. I commented on yours as a courtesy but that's where it ends for me.

**Pull request merged**